queue-jdbc/src/main/java/io/kestra/queue/jdbc/client/JdbcQueueClient.java

```diff
@@ -0,0 +1,198 @@
+        return dslContextWrapper.transactionResult(conf -> {
+            DSLContext context = DSL.using(conf);
+
+            SelectConditionStep<Record> select = context.select(DSL.asterisk())
+                .from(this.jdbcRepository.getTable())
+                .where(io.kestra.jdbc.repository.AbstractJdbcRepository.field("type").eq(queueNameToType(queue)));
+
+            if (routingKey != null) {
+                select = select.and(io.kestra.jdbc.repository.AbstractJdbcRepository.field("routing_key").eq(routingKey));
+            }
+
+            List<JdbcQueueItem> queueItems = select
+                .orderBy(io.kestra.jdbc.repository.AbstractJdbcRepository.field("offset").asc())
+                .limit(configuration.pollSize())
+                .forUpdate()
+                .skipLocked()
+                .fetchInto(JdbcQueueItem.class);
+
+            if (!queueItems.isEmpty()) {
+                List<Long> processedItems = queueItems
+                    .stream()
+                    .map(queueItem -> {
+                        Exception exception = consumer.apply(queueItem.getValue());
+                        return exception == null ? queueItem.getOffset() : null;
+                    })
+                    .filter(Objects::nonNull)
+                    .toList();
+
+                if (!processedItems.isEmpty()) {
+                    DeleteConditionStep<Record> delete = context.delete(this.jdbcRepository.getTable())
+                        .where(io.kestra.jdbc.repository.AbstractJdbcRepository.field("type").eq(queueNameToType(queue)))
+                        .and(io.kestra.jdbc.repository.AbstractJdbcRepository.field("offset", Long.class).in(processedItems));
+
+                    if (routingKey != null) {
+                        delete = delete.and(io.kestra.jdbc.repository.AbstractJdbcRepository.field("routing_key").eq(routingKey));
+                    }
+
+                    delete.execute();
+                }
+            }
+
+            return queueItems.size();
+        });
+    }
+
+    public AtomicReference<Long> subscribeBroadcastMaxOffset(String queue) {
+        AtomicReference<Long> maxOffset = new AtomicReference<>();
+
+        Long initialOffset = dslContextWrapper.transactionResult(conf -> {
+            DSLContext context = DSL.using(conf);
+
+            return context.select(DSL.max(io.kestra.jdbc.repository.AbstractJdbcRepository.field("offset")))
+                .from(this.jdbcRepository.getTable())
+                .where(io.kestra.jdbc.repository.AbstractJdbcRepository.field("type").eq(queueNameToType(queue)))
+                .fetchAny("max", Long.class);
+        });
+
+        if (initialOffset != null) {
+            maxOffset.set(initialOffset);
+        }
+
+        return maxOffset;
+    }
+
+    public Integer subscribeBroadcast(String queue, AtomicReference<Long> maxOffset, MessageConsumer<String, Exception> consumer) {
```

Comment:
Passing an `AtomicReference` as a method param looks like a code smells :) I didn't check yet how that method is used, bere there is probably something better to be done like returning the max offset of the method return.